google-cloud-storage/acceptance/storage/bucket_uniform_bucket_level_access_test.rb

```diff
@@ -198,4 +199,22 @@
     file_default_acl.reload!
     _(file_default_acl.acl.readers).must_equal ["allUsers"]
   end
+
+  it "sets public_access_prevention to enforced and is unable to modify bucket ACL rules" do
+    refute bucket.public_access_prevention_enforced?
+    bucket.public_access_prevention = "enforced"
+    assert bucket.public_access_prevention_enforced?
+    err = expect do
+      bucket.acl.public!
+    end.must_raise Google::Cloud::FailedPreconditionError
+  end
+
+  it "sets enforced public_access_prevention to nil and is able to modify bucket ACL rules" do
+    refute bucket.public_access_prevention_enforced?
+    bucket.public_access_prevention = "enforced"
+    assert bucket.public_access_prevention_enforced?
+    bucket.public_access_prevention = "unspecified"
+    refute bucket.public_access_prevention_enforced?
+    bucket.acl.public!
+  end
 end
```

Comment:
There are a couple test cases from the doc linked from #8154 that I don't see covered:
* Make sure that incorrect values for PAP return 400.
* Test interaction with UBLA (since both are set in the IAM config it's important to check that setting one doesn't inadvertently set/unset the other).

Could you add these?

Reply:
Thanks for catching this! I'm not sure how I missed adding these tests. :( Added now.